site/src/sphinx/server-annotated-service.rst

```diff
@@ -662,6 +732,75 @@ class-level decorators and method-level decorators.
     sb.annotatedService(new MyAnnotatedServiceClass(),
                         new MyGlobalDecorator1());      // order 1
 
+The first rule is as explained before. However, if your own decorator annotations and `@Decorator`_ annotations
+are specified in a mixed order like below, you need to clearly specify their order using ``order()`` attribute
+of the annotation. In the following example, you cannot make sure in what order they decorate the service
+because Java collects repeatable annotations like `@Decorator`_ into a single container annotation like
+`@Decorators`_ so it does not know the specified order between `@Decorator`_ and `@LoggingDecorator`_.
+
+.. code-block:: java
+
+    public class MyAnnotatedServiceClass {
+
+        @Get("/hello")
+        @Decorator(MyMethodDecorator1.class)
+        @LoggingDecorator
+        @Decorator(MyMethodDecorator2.class)
+        public HttpResponse hello() { ... }
+    }
+
+``order()`` attribute makes them in order. It will work only for class-level and method-level decorators.
+The default value of ``order()`` attribute is ``0`` and the decorators are sorted in an ascending order of
+``order()`` value.
+
+.. code-block:: java
+
+    public class MyAnnotatedServiceClass {
+
+        @Get("/hello")
+        @Decorator(value = MyMethodDecorator1.class, order = 1)
+        @LoggingDecorator(order = 2)
+        @Decorator(value = MyMethodDecorator2.class, order = 3)
+        public HttpResponse hello() { ... }
+    }
+
+    // Global-level decorators will not be affected by 'order'.
+    sb.annotatedService(new MyAnnotatedServiceClass(),
+                        new MyGlobalDecorator1());
+
+Even if you want to make a method-level decorator execute first, you can adjust decorator's order as follows.
+
+.. code-block:: java
+
+    @LoggingDecorator
+    public class MyAnnotatedServiceClass {
+
+        // @LoggingDecorator -> MyMethodDecorator1
+        @Get("/hello1")
+        @Decorator(MyMethodDecorator1.class)
+        public HttpResponse hello1() { ... }
+
+        // MyMethodDecorator1 -> @LoggingDecorator
+        @Get("/hello2")
+        @Decorator(value = MyMethodDecorator1.class, order = -1)
+        public HttpResponse hello2() { ... }
+    }
+
+As you read before, you can write your own decorator annotation like `@LoggingDecorator`_.
+Because the order of decorator evaluation depends on the value of an ``order()`` attribute,
+it is recommended to add an ``order()`` attribute to a custom decorator annotation:
```

Comment:
How about:

    If you built a custom decorator annotation like `@LoggingDecorator`_, it is recommended to
    add an ``order()`` attribute so that the user of the custom annotarion is able to adjust
    the order value of the decorator: